airbyte-cdk/bulk/core/load/src/testFixtures/kotlin/io/airbyte/cdk/test/util/IntegrationTest.kt

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk.test.util
+
+import io.airbyte.cdk.command.ConfigurationJsonObjectBase
+import io.airbyte.cdk.command.DestinationCatalog
+import io.airbyte.cdk.command.DestinationStream
+import io.airbyte.cdk.message.DestinationMessage
+import io.airbyte.protocol.models.v0.AirbyteMessage
+import io.airbyte.protocol.models.v0.AirbyteStreamStatusTraceMessage
+import io.airbyte.protocol.models.v0.AirbyteStreamStatusTraceMessage.AirbyteStreamStatus
+import io.airbyte.protocol.models.v0.AirbyteTraceMessage
+import io.airbyte.protocol.models.v0.StreamDescriptor
+import io.micronaut.test.extensions.junit5.annotation.MicronautTest
+import java.time.Instant
+import java.time.LocalDateTime
+import java.time.ZoneOffset
+import java.time.format.DateTimeFormatter
+import java.util.concurrent.atomic.AtomicBoolean
+import javax.inject.Inject
+import kotlin.test.fail
+import kotlinx.coroutines.async
+import kotlinx.coroutines.runBlocking
+import org.apache.commons.lang3.RandomStringUtils
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.parallel.Execution
+import org.junit.jupiter.api.parallel.ExecutionMode
+
+@MicronautTest
+@Execution(ExecutionMode.CONCURRENT)
+abstract class IntegrationTest(
+    val dataDumper: DestinationDataDumper,
+    val destinationCleaner: DestinationCleaner,
+    val recordMangler: ExpectedRecordMapper = NoopExpectedRecordMapper,
+    val nameMapper: NameMapper = NoopNameMapper,
+) {
+    // Intentionally don't inject the actual destination process - we need a full factory
+    // because some tests want to run multiple syncs, so we need to run the destination
+    // multiple times.
+    @Inject lateinit var destinationProcessFactory: DestinationProcessFactory
+
+    private val randomSuffix = RandomStringUtils.randomAlphabetic(4)
+    private val timestampString =
+        LocalDateTime.ofInstant(Instant.now(), ZoneOffset.UTC)
+            .format(DateTimeFormatter.ofPattern("YYYYMMDD"))
+    // stream name doesn't need to be randomized, only the namespace.
+    val randomizedNamespace = "test$timestampString$randomSuffix"
```

Comment:
Does this mean that we never test the null namespace case?

Reply:
we can still do that. Something along these lines, specifics of `config.setDefaultNamespace` tbd:
```kotlin
val modifiedConfig = config.setDefaultNamespace(randomizedNamespace)
runSync(modifiedConfig, records = listOf(Record(namespace=null, ...)))
verify(namespace=randomizedNamespace, ...)
```

Reply:
though it's a bit trickier for s3, where namespace=null is an actual thing :/ I think we'd just disregard `randomizedNamespace` for that:
```kotlin
runSync(config, records=[Record(namespace = null, name = "test_stream_with_random_chars1234")])
```

(i.e. this field is purely advisory, it's up to each individual test to use it or not)